Readme.rmd



CleanAccesions.sh
-Script for extracting the lines of genes from the sorted accesion mapping files, used in loopCleanAccesion.sh
USAGE: CleanAccesions.sh geneList.txt AccessionGene.sorted.map.csv genes.position.csv

CleanFasttree.sh
-Script to run Fasttree inside of HAQESAC to remove trichotomy errors that FastTree will generate, output name will be derived from input
USAGE: fasttree -boot bootstraps -seed seed aln.fas 

CreaterCNVrowNumber.sh
-Search through table generated by diplodocus and extract row numbers for GOI, needed for reading into Rscript.
USAGE: CreaterCNVrowNumber.sh genes.txt Accesssion.sort Diplo.tbl output

DupGroup.py
-Python scrip for reading in a list of groups genereated by findGroupModified.py and mapping possible duplication points onto the species tree. 
USAGE: python3 DupGroup.py speciesTree groups output

Extract6Snakes.sh
-extract all sequences from relevlant snakes from all fastas in a folder. Note: requires single line fasta format
USAGE: Extract6Snakes.sh dir 

FastToScaffAll.py
Read a fasta file and calculate the number of genes included on a single scaffold. Will only work for species that the extraction of gff compatible accession has been done  (usefull in refining search for tandem genes)
USAGE: python3 FastToScaffAll.py Gff fasta SpeciesCode

FastToScaffold.py
-LEGACY: prototype of FastToScaffAll.py without consideration of different formats of accessions.
USAGE: python3 FastToScaffold.py Gff fasta SpeciesCode

FastToScaffoldHYDCUR.py
-LEGACY: prototype of FastToScaffAll.py addapting to HYDCUR accesions 
USAGE: python3 FastToScaffoldHYDCUR.py Gff fasta SpeciesCode

FindDuplicates.py
-Search through a fasta file and find any duplicate in sequence but different inside
USAGE: python3 FindDuplicates.py inFasta outFasta

GffDuplicateSearch.py
search through a proteome and remove multiple proteins encoded by the same gene leaving only one behind. 
USAGE: python3 GffDuplicateSearch.py geneFile RnaFile CDS File fasta output

LoopGffRemove.sh
-Loop through all snakes and execute GffDuplicateSearch.py
USAGE: LoopGffRemove.sh

LoopScaffoldCount.sh
-LEGACY: generate a summary of the scaffold count generated by FastToScaffAll.py for plotting in R
USAGE: LoopScaffoldCount.sh

MSAFolder.sh
-perform a muliple sequence alignment of all fastas in current directory  using clustalo
USAGE: MSAFolder.sh outDir

MapAccessionToGene.py
-create a csv file of all genes and link them to Accession for easy reading by other programs.
USAGE: python3 MapAccessionToGene.py Gff Out SpeciesCode

NCBIToBABSconvert.py
-Read a fasta in NCBI format and convert it to the babs format using a accesion map generated by MapAccessionToGene.py.
USAGE: python3 fasta mapping output

PlotCount.R
-R script to generate the plots used in Thesis

PrePostGene.py
-LEGACY: -Find genes that are upstream and downstream and check if they are present in a fasta converting results into a BED
USAGE: python3 PrePostGene.py fasta CROVV.map NAJNA.map NOTSC.map PSETE.map HYDCUR.map BOACO.map prefix 

PrePostScript.sh
-LEGACY: - run PrePostGene.py across all fastas in a directory and extract fasta from beds generated.
USAGE: PrePostScript.sh inputDir output genomeDir accesionMapDir

RemoveIsoforms.py
-LEGACY: attempted to remove isoforms using python (easier to do using shell script instead)
USAGE: RemoveIsoforms.py inputFasta outputFasta

SAAGATandem.py
-Use the results of MMSeq search of proteome to look for tandem genes output will list one file with all tandem pairs and a second list of all genes. (note search uses strict definition of tandem of immediatly upstream)
USAGE: python3 SAAGATandem.py accesionMap MMSEQ.tbl Prefix

ScaffoldCountAnalysis.sh
-summarise scaffold count into single file for R scripts
USAGE: ScaffoldCountAnalysis.sh

ScaffoldCountPlot.py
-for each snake count the number of scaffolds 
USAGE: python3 ScaffoldCountPlot.py *sorted.csv

TandemCheck.py
-create a CSV shwoing wichi genes are in tandem or on the edege of a scaffold 
USAGE python3 TandemCheck.py fasta CROVV.map NAJAN/map NOTSC.map PSETE.map HYDCUR.map BOACO.map outfile

TreeMakeFolder.sh
-make a indiviudal clustal tree for eah file in a folder
USAGE: TreeMakeFolder.sh outDir

UniqeCount.py
-Count the number of protiens from each species in a fasta, divided into taxonomy groups 
USAGE: python3 UniqeCount.py fasta

backup.sh
copy given folders to a backup directory 
-USAGE backup.sh dir

combineVenomControlTree.py
-subtact the control tree duplication rate form the venom tree
USAGE: python3 combineVenomControlTree.py venomtree controltree outFasta

dateLoop.sh
- loop through and add date prefix to a file 
USAGE: dateLoop.sh Files 

fastaToBed.py
- convert all protines in a fasta file to bed format to extract genomic positions
USAGE: python3 fastaToBed.py fasta CROVV.map NAJAN/map NOTSC.map PSETE.map HYDCUR.map BOACO.map outfile

filterIsoforms.py
LEGACY-filter files for isofroms eaving only one and a log of which are removed 
USAGE: pyhton3 filterIsoforms.py inFasta outFasta


filterNAJNA_HYDCUR.py
LEGACY-filter NAJAN and HYDCUR for isofroms eaving only one and a log of which are removed 
USAGE python3 filterNAJNA_HYDCUR.py inFasta outFasta

findGroupModified.py
-Find all duplicaton points in a nwk tree create group file
USAGE: pyhton3 findGroupModified.py nwk

formatGff.sh
-split a gff into CDS, RNA and genes
USAGE: formatGff.sh *.gff

gff2prot.pl
-extract protiens from a gff
USAGE gff2prot.pl genome gff outname

groupAl.py
LEGACY- find subfamiies in a tree based on QFO human and mouse genes 
USAGE: pyhton3 groupAl.py nwk

groupings.py
-LEGACY: find gorupoings in a nwk tree
USAGE: pyhton3 groupings.py nwk

loopCleanAccesion.sh
-Loop CleanAccesions.sh for each species 
USAGE: loopCleanAccesion.sh

loopCounts.sh
-loop UniqeCount.py for every fasta in curretn directory
USAGE: loopCounts.sh outfile

loopFastToBed.sh
-loop fastaToBed for every fasta in curretn directory
USAGE: loopFastToBed.sh indir outdir mappeddir

loopFastToScaffs.sh
-loop FastToScaffAll.py for every file in a list 
USAGE: loopFastToScaffs.sh list.txt

loopFindGroup.sh
-loop findGroupModified.py for every file in a directory
USAGE: loopFindGroup.sh inDir outDir

loopSAGAATandem.sh
-loop SAAGATandem.py for every snake 
USAGE: loopSAGAATandem.sh

loopTandem.sh
-loop TandemCheck.py for every fasta in a directory 
USAGE: loopTandem.sh inDir outDir mappeddir

loopTandemVisual.sh
- loop through programs to find extract and plot visualisation of tandem regions 
USAGE: loopTandemVisual.sh inDir outDir mappeddir genomeDir

prefDate.sh
-prefix every subfile with current date
USAGE: prefDate.sh

removeTrichotomy.py
-remove trichotomy from a nwk file 
USAGE: removeTrichotomy.py inNwk outNwk

renammeScaffs.sh
-Use NCBI to BABS mapping file to rename scaffolds
USAGE: renammeScaffs.sh mapping gff

runGepard.sh
-create visualisation of a genome region from fasta
USAGE: runGepard.sh inDir lower outDir

scaffoldCords.py
- extrect tandem regions using 150000kb limit
USAGE: scaffoldCords.py bed outfile

sortAccession.py
-Sort a mapping file by accesion
USAGE: python3 sortAccession.py mapping outfile

specCount.sh
-Count numbers of proteins from each species in a fasta file 
USAGE: specCount.sh fasta

splitFasta.sh
-Split a fasta file into multiple files
USAGE: splitFasta.sh fasta outPrefix

tarball.sh
-create a tarball of a file  
USAGE: tarball.sh file

tree.py
-print a nwk tree
USAGE: python3 tree.py nwk

tricordFix.sh
LEGACY- try to remove trichotomy using unix tools
USAGE: tricordFix.sh tree