Readme.rmd



CleanAccesions.sh
-Script for extracting the lines of genes from the sorted accesion mapping files, used in loopCleanAccesion.sh
USAGE: CleanAccesions.sh geneList.txt AccessionGene.sorted.map.csv genes.position.csv

CleanFasttree.sh
-Script to run Fasttree inside of HAQESAC to remove trichotomy errors that FastTree will generate, output name will be derived from input
USAGE: fasttree -boot bootstraps -seed seed aln.fas 

CreaterCNVrowNumber.sh
-Search through table generated by diplodocus and extract row numbers for GOI, needed for reading into Rscript.
USAGE: CreaterCNVrowNumber.sh genes.txt Accesssion.sort Diplo.tbl output


DupGroup.py
-Python scrip for reading in a list of groups genereated by findGroupModified.py and mapping possible duplication points onto the species tree. 
USAGE: python3 DupGroup.py speciesTree groups output

Extract6Snakes.sh
-extract all sequences from relevlant snakes from all fastas in a folder. Note: requires single line fasta format
USAGE: Extract6Snakes.sh dir 

FastToScaffAll.py
Read a fasta file and calculate the number of genes included on a single scaffold. Will only work for species that the extraction of gff compatible accession has been done  (usefull in refining search for tandem genes)
USAGE: python3 FastToScaffAll.py Gff fasta SpeciesCode

FastToScaffold.py
-LEGACY: prototype of FastToScaffAll.py without consideration of different formats of accessions.
USAGE: python3 FastToScaffold.py Gff fasta SpeciesCode

FastToScaffoldHYDCUR.py
-LEGACY: prototype of FastToScaffAll.py addapting to HYDCUR accesions 
USAGE: python3 FastToScaffoldHYDCUR.py Gff fasta SpeciesCode

FindDuplicates.py
-Search through a fasta file and find any duplicate in sequence but different inside
USAGE: python3 FindDuplicates.py inFasta outFasta

GffDuplicateSearch.py
search through a proteome and remove multiple proteins encoded by the same gene leaving only one behind. 
USAGE: python3 GffDuplicateSearch.py geneFile RnaFile CDS File fasta output

LoopGffRemove.sh
-Loop through all snakes and execute GffDuplicateSearch.py
USAGE: LoopGffRemove.sh

LoopScaffoldCount.sh
-LEGACY: generate a summary of the scaffold count generated by FastToScaffAll.py for plotting in R
USAGE: LoopScaffoldCount.sh

MSAFolder.sh
MapAccessionToGene.py
NCBIToBABSconvert.py
PlotCount.R
PrePostGene.py
PrePostScript.sh
RemoveIsoforms.py
SAAGATandem.py
ScaffoldCountAnalysis.sh
ScaffoldCountPlot.py
TandemCheck.py
TreeMakeFolder.sh
UniqeCount.py
backup.sh
combineVenomControlTree.py
dateLoop.sh
fastaToBed.py
filterIsoforms.py
filterNAJNA_HYDCUR.py
findGroupModified.py
formatGff.sh
gff2prot.pl
groupAl.py
groupings.py
loopCleanAccesion.sh
loopCounts.sh
loopFastToBed.py
loopFastToBed.sh
loopFastToScaffs.sh
loopFindGroup.sh
loopSAGAATandem.sh
loopTandem.sh
loopTandemVisual.sh
prefDate.sh
removeTrichotomy.py
renammeScaffs.sh
runGepard.sh
scaffoldCords.py
sortAccession.py
specCount.sh
splitFasta.sh
tarball.sh
tree.py
tricordFix.sh